x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/endpoint_details_tabs.tsx

```diff
@@ -24,53 +29,80 @@ interface EndpointDetailsTabs {
   content: JSX.Element;
 }
 
-const StyledEuiTabbedContent = styled(EuiTabbedContent)`
-  overflow: hidden;
-  padding-bottom: ${(props) => props.theme.eui.paddingSizes.xl};
-
-  > [role='tabpanel'] {
-    height: 100%;
-    padding-right: 12px;
-    overflow: hidden;
-    overflow-y: auto;
-    ::-webkit-scrollbar {
-      -webkit-appearance: none;
-      width: 4px;
-    }
-    ::-webkit-scrollbar-thumb {
-      border-radius: 2px;
-      background-color: rgba(0, 0, 0, 0.5);
-      -webkit-box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
-    }
-  }
-`;
-
 export const EndpointDetailsFlyoutTabs = memo(
-  ({ show, tabs }: { show: EndpointIndexUIQueryParams['show']; tabs: EndpointDetailsTabs[] }) => {
+  ({
+    hostname,
+    show,
+    tabs,
+  }: {
+    hostname?: string;
+    show: EndpointIndexUIQueryParams['show'];
+    tabs: EndpointDetailsTabs[];
+  }) => {
+    const dispatch = useDispatch<(action: EndpointAction) => void>();
+    const { pageSize } = useEndpointSelector(getActivityLogDataPaging);
     const [selectedTabId, setSelectedTabId] = useState<EndpointDetailsTabsId>(() => {
       return show === 'details'
         ? EndpointDetailsTabsTypes.overview
         : EndpointDetailsTabsTypes.activityLog;
     });
 
     const handleTabClick = useCallback(
-      (tab: EuiTabbedContentTab) => setSelectedTabId(tab.id as EndpointDetailsTabsId),
-      [setSelectedTabId]
+      (tab: EuiTabbedContentTab) => {
+        dispatch({
```

Comment:
From our side conversation:
The switch between tabs should be triggered via the URL by setting the `show` property. From a middleware/reducer standpoint, we should have a selector in place that checks the `show` param value to determine what tabs/sub-view they are in while the floyout is open.

Reply:
I actually wanted to do that. But there was added complexity of setting the `show` URL search param. Right now when details tab is open there's no `show=details` in URL params. There's a `show=policy_response` when we open the policy response panel though. 

Now, if you select another endpoint, it removes the `show=` param and shows the endpoint details flyout instead of the policy response panel. I've created a ticket for this. https://github.com/elastic/security-team/issues/1344

Reply:
Here's a ticket for doing this https://github.com/elastic/security-team/issues/1346